node/src/BaseClient.ts

```diff
@@ -3197,13 +3206,14 @@ export class BaseClient {
         return this.createWritePromise(createXLen(key));
     }
 
-    private readonly MAP_READ_FROM_STRATEGY: Record<
```

Comment:
I'm missing some context here. Why do we want to change from Record to Map? I'm not really sure I understand the differences between them either.

Reply:
This doesn't work with enum or I don't know now to make this work with an enum. Note, `ReadFrom` changed from type to enum.

Reply:
I'm also confused by that too. Do we want to change from a union to an enum? https://stackoverflow.com/questions/40275832/typescript-has-unions-so-are-enums-redundant seems to suggest that unions are preferred.